packages/typespec-ts/test/unit/clientFactoryGenerator.spec.ts

```diff
@@ -302,44 +305,19 @@ describe("Client Factory generation", () => {
         }]>)
       @service( {title: "PetStoreClient"})
       namespace PetStore;
-      `);
-      assert.ok(models);
-      assertEqualContent(
-        models!.content,
-        `
-        import { getClient, ClientOptions } from "@azure-rest/core-client";
-        import { logger } from "./logger";
-        import { testClient } from "./clientDefinitions";
-        
-        /**
-         * Initialize a new instance of \`testClient\`
-         * @param endpoint - The parameter endpoint
-         * @param options - the parameter for all optional parameters
-         */
-        export default function createClient(endpoint: string, options: ClientOptions = {}): testClient {
-        const baseUrl = options.baseUrl ?? \`\${endpoint}\`;
-        
-        const userAgentInfo = \`azsdk-js--rest/1.0.0-beta.1\`;
-        const userAgentPrefix =
-            options.userAgentOptions && options.userAgentOptions.userAgentPrefix
-            ? \`\${options.userAgentOptions.userAgentPrefix} \${userAgentInfo}\`
-            : \`\${userAgentInfo}\`;
-        options = {
-            ...options,
-            userAgentOptions: {
-            userAgentPrefix,
-            },
-            loggingOptions: {
-              logger: options.loggingOptions?.logger ?? logger.info
-            },
-        };
-        
-        const client = getClient(baseUrl, options) as testClient;
-        
-        return client;
-    }
-    `
-      );
+      `,
+          false,
+          false
+        );
+        assert.fail("Should throw diagnostic errors");
```

Comment:
Is it come from anywhere that we should throe diagnostic errors ?

Reply:
https://github.com/Azure/autorest.typescript/pull/1924/files/3ab05f96704bfa7e4ff074e474e63808e1e23963#r1265061076